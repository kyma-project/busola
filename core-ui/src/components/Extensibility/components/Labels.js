import { useGetPlaceholder } from 'components/Extensibility/helpers';
import { isNil } from 'lodash';
import { Labels as BusolaLabels } from 'shared/components/Labels/Labels';
import { Tokens } from 'shared/components/Tokens';

export function Labels({ schema, value, structure }) {
  const { emptyLeafPlaceholder } = useGetPlaceholder(structure);

  const isSimpleType = value => {
    return (
      typeof value === 'boolean' ||
      typeof value === 'string' ||
      typeof value === 'number'
    );
  };

  switch (true) {
    case isNil(value):
      return emptyLeafPlaceholder;
    case Array.isArray(value):
      return <Tokens tokens={value} />;
    case isSimpleType(value):
      return value;
    default:
      return <BusolaLabels labels={value} />;
  }
}

Labels.inline = true;
Labels.array = true;
