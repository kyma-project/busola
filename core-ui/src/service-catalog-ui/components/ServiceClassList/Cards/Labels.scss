.fd-panel__footer {
  &.service-list--card__footer {
    border-top: none;
    justify-content: flex-start;
    padding: 0;
    &__labels {
      display: inline-block;
      margin: 8px 8px 0 0;
    }
  }
}
