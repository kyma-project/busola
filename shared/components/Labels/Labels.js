import React from 'react';
import { EMPTY_TEXT_PLACEHOLDER } from '../../constants/constants';

export const Labels = labels => {
  labels = labels.labels;
  if (!labels || Object.keys(labels).length === 0) {
    return <span>{EMPTY_TEXT_PLACEHOLDER}</span>;
  }
  const separatedLabels = [];
  /* eslint-disable no-unused-vars */
  for (const key in labels) {
    separatedLabels.push(`${key}=${labels[key]}`);
  }

  /* eslint-enable no-unused-vars */
  return separatedLabels.map((label, id) => (
    <span
      aria-label={label}
      className="fd-token fd-token--readonly "
      key={id}
      style={{ marginBottom: '4px', marginRight: '4px' }}
    >
      <span className="fd-token__text fd-has-font-size-small">{label}</span>
    </span>
  ));
};
