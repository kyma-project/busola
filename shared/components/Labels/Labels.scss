.labels {
  display: flex;
  flex-wrap: wrap;

  .fd-token {
    margin-bottom: 4px;
    margin-right: 4px;
    white-space: normal;
    height: auto;
    line-height: 1rem;
    padding: 0.25rem 0.4rem !important;
  }
}
