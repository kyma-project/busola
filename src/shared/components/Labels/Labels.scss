.labels {
  display: grid;
  justify-items: start;
}

.token {
  margin-top: 4px;
  margin-bottom: 4px;
}
