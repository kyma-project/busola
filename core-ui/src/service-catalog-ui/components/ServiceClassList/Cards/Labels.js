import React from 'react';
import { Token } from 'fundamental-react';
import { Tooltip } from 'shared/components/Tooltip/Tooltip';
import './Labels.scss';
import { isStringValueEqualToTrue } from 'helpers';
const labelsDescription = {
  'connected-app':
    'This Service Class is connected to a given application by Application Connector.',
  showcase:
    'This Service Class presents a specific functionality. Do not use it on the production cluster.',
};
export function Labels({ labels, ignoredLabels = [] }) {
  const ALWAYS_IGNORED_LABELS = ['local', 'provisionOnlyOnce'];
  return (
    <div className="service-list--card__footer">
      {labels &&
        Object.keys(labels).map(label => {
          if (
            ALWAYS_IGNORED_LABELS.includes(label) ||
            ignoredLabels.includes(label)
          ) {
            return null;
          }
          if (
            (label === 'showcase' &&
              !isStringValueEqualToTrue(labels[label])) ||
            (label === 'connected-app' && !labels[label])
          ) {
            return null;
          }

          return (
            <div className="service-list--card__footer__labels" key={label}>
              <Tooltip content={labelsDescription[label] || ''}>
                <Token readOnly buttonLabel="">
                  {label === 'connected-app' ? labels['connected-app'] : label}
                </Token>
              </Tooltip>
            </div>
          );
        })}
    </div>
  );
}
