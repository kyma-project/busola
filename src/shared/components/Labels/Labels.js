import React from 'react';
import classNames from 'classnames';
import { EMPTY_TEXT_PLACEHOLDER } from 'shared/constants';
import { Badge, Label } from '@ui5/webcomponents-react';

import { spacing } from '@ui5/webcomponents-react-base';
import './Labels.scss';
import { useTranslation } from 'react-i18next';

const SHORTENING_TRESHOLD = 50;

export const Labels = ({
  labels,
  className = '',
  shortenLongLabels = true,
  style = null,
  displayLabelForLabels = false,
}) => {
  const { t } = useTranslation();
  if (!labels || Object.keys(labels).length === 0) {
    return (
      <>
        {displayLabelForLabels ? (
          <Label showColon={true} style={spacing.sapUiTinyMarginBottom}>
            {t('common.headers.labels')}
          </Label>
        ) : null}
        <div>{EMPTY_TEXT_PLACEHOLDER}</div>
      </>
    );
  }
  const separatedLabels = [];
  for (const key in labels) {
    separatedLabels.push(`${key}=${labels[key]}`);
  }

  const shortenLabel = label => label.slice(0, SHORTENING_TRESHOLD) + '...';

  return (
    <>
      {displayLabelForLabels ? (
        <Label
          showColon={true}
          style={spacing.sapUiTinyMarginBottom}
          className="bsl-has-color-status-4"
        >
          {t('common.headers.labels')}
        </Label>
      ) : null}
      <div className={classNames('labels', className)} style={style}>
        {separatedLabels.map((label, id) => (
          <Badge
            aria-label={label}
            key={id}
            colorScheme="10"
            style={{
              ...spacing.sapUiTinyMarginEnd,
              ...spacing.sapUiTinyMarginBottom,
            }}
          >
            {shortenLongLabels && label.length > SHORTENING_TRESHOLD
              ? shortenLabel(label)
              : label}
          </Badge>
        ))}
      </div>
    </>
  );
};
