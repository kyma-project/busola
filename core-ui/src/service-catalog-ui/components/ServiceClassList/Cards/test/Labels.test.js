import { render } from 'enzyme';
import React from 'react';
import { Labels } from '../Labels';
import { expectKnownConsoleWarnings } from '../../../../testing';

const consoleWarn = jest.spyOn(global.console, 'warn').mockImplementation();
afterAll(() => {
  consoleWarn.mockReset();
});
describe('Labels', () => {
  it('Render no labels', () => {
    const component = render(<Labels labels={null} />);
    expect(component).toMatchSnapshot();
  });
  it('Render empty labels', () => {
    const component = render(<Labels labels={{}} />);
    expect(component).toMatchSnapshot();
  });
  it('Render skips "local"', () => {
    const component = render(
      <Labels labels={{ local: 'true', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Render skips "provisionOnlyOnce"', () => {
    const component = render(
      <Labels labels={{ provisionOnlyOnce: 'true', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Render keeps "true" "showcase', () => {
    const component = render(
      <Labels labels={{ showcase: 'true', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Render keeps "TrUe" "showcase', () => {
    const component = render(
      <Labels labels={{ showcase: 'TrUe', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Render skips with not "true" "showcase', () => {
    const component = render(
      <Labels labels={{ showcase: 'false', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Render skips empty "connected-app"', () => {
    const component = render(
      <Labels labels={{ 'connected-app': '', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Render value of "connected-app"', () => {
    const component = render(
      <Labels labels={{ 'connected-app': 'application', random: 'value' }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
