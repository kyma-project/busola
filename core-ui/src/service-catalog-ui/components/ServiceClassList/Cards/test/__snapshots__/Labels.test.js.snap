// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels Render empty labels 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
/>
`;

exports[`Labels Render keeps "TrUe" "showcase 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          showcase
        </span>
      </div>
    </div>
  </div>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Labels Render keeps "true" "showcase 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          showcase
        </span>
      </div>
    </div>
  </div>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Labels Render no labels 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
/>
`;

exports[`Labels Render skips "local" 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Labels Render skips "provisionOnlyOnce" 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Labels Render skips empty "connected-app" 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Labels Render skips with not "true" "showcase 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Labels Render value of "connected-app" 1`] = `
<div
  class="fd-panel__footer service-list--card__footer"
>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          application
        </span>
      </div>
    </div>
  </div>
  <div
    class="service-list--card__footer__labels"
  >
    <div
      class=""
      style="display:inline"
    >
      <div
        class="sc-ckVGcZ fRGHMA"
      >
        <span
          class="fd-token sc-jKJlTe fEwmSA"
          role="button"
        >
          random
        </span>
      </div>
    </div>
  </div>
</div>
`;
